### PR TITLE
Android: Convert InputOverlay to Kotlin

### PR DESCRIPTION
**Pull request by t895**

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt [hunk 1]
```diff
@@ -0,0 +1,2347 @@
...
+                            dpadPressed[3]
+                        )
+                    }
+                }
+
+                MotionEvent.ACTION_UP,
+                MotionEvent.ACTION_POINTER_UP -> {
+                    // If a pointer ends, release the buttons.
+                    if (dpad.trackId == event.getPointerId(pointerIndex)) {
+                        for (i in 1 until 4) {
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt:253]: Not from 0?

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt [hunk 2]
```diff
@@ -0,0 +1,2347 @@
...
+            .putFloat(getXKey(sharedPrefsId, controllerType, orientation), x.toFloat())
+            .putFloat(getYKey(sharedPrefsId, controllerType, orientation), y.toFloat())
+            .apply()
+    }
+
+    /**
+     * Initializes an InputOverlayDrawableButton, given by resId, with all of the
+     * parameters set for it to be properly shown on the InputOverlay.
+     *
+     *
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt:1070]: Probably replace these double blank lines with single blank lines.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        val pointerIndex = if (firstPointer) 0 else event.actionIndex
+        var pressed = false
+
+        when (action) {
+            MotionEvent.ACTION_DOWN,
+            MotionEvent.ACTION_POINTER_DOWN -> {
+                if (bounds.contains(
+                        event.getX(
+                            pointerIndex
+                        ).toInt(), event.getY(pointerIndex).toInt()
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt:111]: This kind of stuff kind of hurts to read compared to the Java formatting... But if it has to be formatted that way in the standard way of formatting Kotlin, I guess we'll go with it.

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayPointer.kt [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    private fun updateOldAxes() {
+        oldX = x
+        oldY = y
+    }
+
+    private fun reset() {
+        oldY = 0.0f
+        oldX = oldY
+        y = oldX
+        x = y
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayPointer.kt:122]:
```suggestion
        oldY = 0.0f
        oldX = 0.0f
        y = 0.0f
        x = 0.0f
```

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt (hunk 1 above)
Comment (t895) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlay.kt:253]: My bad, good catch

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt (hunk 3 above)
Comment (t895) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt:111]: That one is pretty bad, but that's only the default formatting. Would this be better?
```suggestion
                if (bounds.contains(
                        event.getX(pointerIndex).toInt(),
                        event.getY(pointerIndex).toInt()
```

Comment (JosJuice) [reply Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/overlay/InputOverlayDrawableJoystick.kt]: Yeah, that's a lot better.

**Pull request merged**